### ISSUE TITLE
Class ifgzstream throws an exception

### ISSUE THREAD

**Issue by v-dobrev** [mfem/mfem#1412]
Class `ifgzstream` (and the inherited class `named_ifgzstream`) throws an exception instead of just setting the `failbit` as `std::ifstream` does.

This causes mysterious errors to be printed when a mesh file is not found instead of printing the message that the mesh file is not found.

**Comment by markcmiller86** [comment]
@tzanio took a quick look. All that code has changed size these classes were updated to avoid GNU licensed gzstream classes I originally employed.

Am guessing but maybe something related to this...

https://github.com/mfem/mfem/blob/278a27db5ed1ccc58253c604024d8388f4bde7a8/general/zstr.hpp#L784

**Comment by v-dobrev** [comment]
@markcmiller86, thanks for checking. You maybe right -- calling `exceptions` (https://en.cppreference.com/w/cpp/io/basic_ios/exceptions) maybe the cause of this. However, I don't see any calls to `exceptions(std::ios_base::failbit)`, so there maybe something else. There are a number of explicit `throw` statements in the file.

Investigating further, on Mac, when I ran `./ex1 -m zzz`, I got this:
```console
libc++abi.dylib: terminating with uncaught exception of type strict_fstream::Exception: strict_fstream: open('zzz',in): open failed: No such file or directory
Abort
```
which is not that bad as an error message. However, on RHEL7, I got this:
```console
terminate called after throwing an instance of 'std::length_error'
  what():  basic_string::resize
Abort (core dumped)
```
which is quite confusing.

In any case, I think it is better to change `ifgzstream` to set `failbit` and not throw exception when opening the file fails.
